rtf_vis_tool/src/Components/frustum_classes/se3.js

```diff
@@ -0,0 +1,58 @@
+/*Contains the Pose Matrix used to convert points from camera coordinate system to world coordinate system.
+
+Author: Adi Singh
+*/
+
+// Third-Party Package Imports.
+var nj = require('numjs');
+
+class SE3 {
+    // An SE3 class allows point cloud rotation and translation operations.
+
+    constructor(rotation, translation) {
+        /*Initialize an SE3 instance and pose matrix with its rotation and translation matrices.
+
+            Args:
+                rotation: Matrix of shape 3x3
+                translation: Array of length 3
+        */
+        if (rotation.shape[0] != 3 || rotation.shape[1] != 3) throw 'Invalid Rotation Matrix';
+        if (translation.shape[0] != [3]) throw 'Invalid Translation Matrix';
+        this.rotation = rotation;
+        this.translation = translation;
+        this.transform_matrix = nj.identity(4);
+
+        //set upper left 3x3 to rotation matrix
+        for (var row = 0; row < 3; row++) {
+            for (var col = 0; col < 3; col++) {
+                this.transform_matrix.set(row, col, rotation.get(row,col));
+            }
+        }
+
+        //set last column to translation array
+        for (var row = 0; row < 3; row++) {
+            this.transform_matrix.set(row, 3, this.translation.get(row));
+        }
+    }
+
+    tranform_frustum_vertex(point_cam_coords) {
```

Comment:
this function name should be more general, since we can use it to transform anything, not just frustum vertices.

So `transform_from()` is the preferred name, per GTSAM conventions https://gtsam.org/gtsam.org/2020/06/28/gtsam-conventions.html